### ISSUE TITLE
Add new Grouped MenuHandler

### ISSUE THREAD

**Issue by crynobone** [orchestral/foundation#182]
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Imagine if you have the following menu structure. 

```
- Helpdesk
-- Ticket
-- Category
```

Using the proposed solution we now can do the following.

```php
class CategoryMenuHandler extends Orchestra\Foundation\Support\MenuHandler
{
    /**
     * Menu configuration.
     *
     * @var array
     */
    protected $menu = [
        'id'       => 'category',
        'title'    => 'Category',
        'link'     => 'orchestra::helpdesk/categories',
    ];

    /**
     * Check authorization to display the menu.
     *
     * @param  \Orchestra\Contracts\Authorization\Authorization  $acl
     *
     * @return bool
     */
    public function authorize(Authorization $acl)
    {
        return true;
    }
}
```

```php
class TicketMenuHandler extends Orchestra\Foundation\Support\MenuHandler
{
    /**
     * Menu configuration.
     *
     * @var array
     */
    protected $menu = [
        'id'       => 'ticket',
        'title'    => 'Ticket',
        'link'     => 'orchestra::helpdesk/tickets',
    ];

    /**
     * Check authorization to display the menu.
     *
     * @param  \Orchestra\Contracts\Authorization\Authorization  $acl
     *
     * @return bool
     */
    public function authorize(Authorization $acl)
    {
        return true;
    }
}
```

```php
class HelpdeskMenuHandler extends Orchestra\Foundation\Support\MenuHandler
{
    /**
     * Menu configuration.
     *
     * @var array
     */
    protected $menu = [
        'id'       => 'helpdesk',
        'position' => '*',
        'title'    => 'Helpdesk',
        'link'     => 'orchestra::helpdesk',
        'with'     => [
            CategoryMenuHandler::class,
            TicketMenuHandler::class,
        ],
    ];

    /**
     * Check authorization to display the menu.
     *
     * @param  \Orchestra\Contracts\Authorization\Authorization  $acl
     *
     * @return bool
     */
    public function authorize(Authorization $acl)
    {
        return $acl->can('manage ticket');
    }
}
```

Using the `with` key, we can attach more menu handler class and set it as child of `HelpdeskMenuHandler`.

**Comment by ahmadshah** [comment]
This one is a +1 from me! :+1:

**Comment by irazasyed** [comment]
This is good :+1:

**Comment by crynobone** [comment]
Implemented in `v3.0.13` and `v3.1.1`.
